src/deprecated.jl

```diff
@@ -1430,6 +1430,12 @@ function Base.getindex(itr::DataFrameColumns{<:SubDataFrame,AbstractVector}, j)
     itr.df[:, j]
 end
 
+function Base.getindex(itr::DataFrameColumns{<:SubDataFrame,AbstractVector}, j::Int)
```

Comment:
A side comment.
In the past we did not have `DataFrameColumns{<:SubDataFrame,AbstractVector}`, as we only had `DataFrameColumns{<:SubDataFrame, Pair{Symbol, AbstractVector}}` (under another name).

Thus `DataFrameColumns{<:SubDataFrame,AbstractVector}` can safely return a view - consistently with the target behavior.

But if for some reason you prefer to add it it is OK. I will remove it in `getindex` cleanup PR later.

Reply:
I just added this to fix an ambiguity error I was getting. I guess I should just remove the `depwarn` call from it.

Reply:
Ah - in this case it can stay or would write `itr.df[j]` below and remove a depwarn (as then depwarn will be triggered by `getindex` in `itr.df[j]` call).

Reply:
OK let's keep it that way, it's simpler if it's identical to the method above.